heroic-component/src/main/java/com/spotify/heroic/cluster/ClusterManager.java

```diff
@@ -75,4 +109,14 @@
     Statistics getStatistics();
 
     Set<RpcProtocol> protocols();
+
+    /**
+     * The result of applying a function to a random node in a shard
+     */
+    @Data
+    class NodeResult<T> {
+        private final T returnValue;
+        private final UUID nodeId;
+        private final String nodeString;
```

Comment:
I like wrapping the result to provide more information, but maybe we should expose the `ClusterNode` directly here?